src/components/views/dialogs/SetMxIdDialog.js

```diff
@@ -101,6 +101,9 @@ export default React.createClass({
     },
 
     onSubmit: function(ev) {
+        if (this.refs.uiAuth) {
+            this.refs.uiAuth.tryContinue();
```

Comment:
Calling methods on child elements is not generally a very react-y pattern: in this case probably a symptom of logic being in the react component rather than its own class. Not suggesting going & factoring it out here though, this is probably fine in this instance.

Reply:
if it helps, it felt very wrong to do this while writing it. Ideally we wouldn't have components depend on the state of each other at all :(